openedx/core/djangoapps/content/search/views.py

```diff
@@ -29,6 +35,26 @@ def _get_meili_api_key_uid():
     return _get_meili_api_key_uid.uid
 
 
+def _get_meili_access_filter(request: Request) -> dict:
+    """
+    Return meilisearch filter based on the requesting user's permissions.
+    """
+    # Global staff can see anything, so no filters required.
+    if GlobalStaff().has_user(request.user):
+        return {}
+
+    # Everyone else is limited to their org roles...
+    user_orgs = [
+        org.short_name for org in get_user_orgs(request.user)[:MAX_ORGS_IN_FILTER]
```

Comment:
If the user can access only one course in `orgA`, `orgA` will be returned here. I think we want to add only orgs that the user is an admin here. Maybe use `_get_content_creator_orgs` ?

Reply:
See [1181eea](https://github.com/openedx/edx-platform/pull/34471/commits/1181eeac843148f2b557eedb166a0143fb0c31a9)